spring-cloud-gcp-pubsub/src/test/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubBatchMessageSourceTests.java

```diff
@@ -0,0 +1,250 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.integration.inbound;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import org.springframework.cloud.gcp.pubsub.core.subscriber.PubSubSubscriberOperations;
+import org.springframework.cloud.gcp.pubsub.integration.AckMode;
+import org.springframework.cloud.gcp.pubsub.support.AcknowledgeablePubsubMessage;
+import org.springframework.cloud.gcp.pubsub.support.GcpPubSubHeaders;
+import org.springframework.integration.IntegrationMessageHeaderAccessor;
+import org.springframework.integration.acks.AcknowledgmentCallback;
+import org.springframework.integration.endpoint.MessageSourcePollingTemplate;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandlingException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for {@link PubSubBatchMessageSource}.
+ *
+ * @author Elena Felder
+ * @since 1.2
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class PubSubBatchMessageSourceTests {
+
+	private PubSubSubscriberOperations mockPubSubSubscriberOperations;
+
+	@Mock
+	private AcknowledgeablePubsubMessage msg1;
+
+	@Mock
+	private AcknowledgeablePubsubMessage msg2;
+
+	@Mock
+	private AcknowledgeablePubsubMessage msg3;
+
+	@Before
+	public void setUp() {
+		this.mockPubSubSubscriberOperations = mock(PubSubSubscriberOperations.class);
+
+		when(this.mockPubSubSubscriberOperations.pull("sub1", 1, true))
+				.thenReturn(Collections.singletonList(this.msg1));
+	}
+
+	@Test
+	public void doReceive_returnsNullWhenNoMessagesAvailable() {
+		when(this.mockPubSubSubscriberOperations.pull("sub1", 1, true))
+				.thenReturn(Collections.emptyList());
+
+		PubSubBatchMessageSource pubSubMessageSource = new PubSubBatchMessageSource(
+				this.mockPubSubSubscriberOperations, "sub1");
+		pubSubMessageSource.setMaxFetchSize(1);
+
+		Object message = pubSubMessageSource.doReceive(1);
+
+		assertThat(message).isNull();
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	public void doReceive_callsPubsubAndCachesCorrectly() {
```

Comment:
```suggestion
	public void doReceive_callsPubsub) {
```

There is no caching in the the batch one.